File: Java/dependency-injection/control-freak-refactored/src/main/java/com/github/grzesiek_galezowski/DependencyInjectionAfter/Outbound/TcpSocket.java
package com.github.grzesiek_galezowski.DependencyInjectionAfter.Outbound;

public class TcpSocket implements ISocket {
  public void Open() {
    System.out.println("open");
  }

  public void Close() {
    System.out.println("closing");
  }

  public void Send(String lol) {
    System.out.println(lol);
  }
}

File: CSharp/DependencyInjection/DependencyInjectionAfter/_1_ControlFreakRefactoredToDependencyInjection/Outbound/TcpSocket.cs
using System;

namespace DependencyInjectionAfter._1_ControlFreakRefactoredToDependencyInjection.Outbound
{
  public interface ISocket
  {
    void Open();
    void Close();
    void Send(string lol);
  }

  public class TcpSocket : ISocket
  {
    public void Open()
    {
      Console.WriteLine("open");
    }

    public void Close()
    {
      Console.WriteLine("closing");
    }

    public void Send(string lol)
    {
      Console.WriteLine(lol);
    }
  }
}